src/Agent.Listener/CapabilitiesProvider.cs

```diff
@@ -0,0 +1,291 @@
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener
+{
+    [ServiceLocator(Default = typeof(CapabilitiesProvider))]
+    public interface ICapabilitiesProvider : IAgentService
+    {
+        Task<Dictionary<string, string>> GetCapabilitiesAsync(string agentName, CancellationToken token);
+    }
+
+    public class Capability
+    {
+        public Capability(string name, string tool = null, string[] paths = null)
+        {
+            Name = name;
+            Tool = tool;
+            Paths = paths;
+        }
+
+        //Name of the capability as requried by the task definition.
+        public string Name { get; private set; }
+
+        //Name of an executable file, which if found in the file system means the capability is present.
+        public string Tool { get; private set; }
+
+        //An array of paths, which if found in the file system means the capability is present.
+        public string[] Paths { get; private set; }
+    }
+
+    public class ToolCapability
+    {
+        public ToolCapability(string name, string command, string commandArgs = null)
+        {
+            Name = name;
+            Command = command;
+            CommandArgs = commandArgs;
+        }
+
+        //Name of the capability as requried by the task definition.
+        public string Name { get; private set; }
+
+        //Name of an executable file, which can determine if a capability is present.
+        //The script is expected to print on the standard output the value for the capability.
+        public string Command { get; private set; }
+
+        //Arguments passed to Command.
+        public string CommandArgs { get; private set; }
+    }
+
+    public sealed class CapabilitiesProvider : AgentService, ICapabilitiesProvider
+    {
+        private List<Capability> _regularCapabilities =
+            new List<Capability>
+            {
+                new Capability( "ant" ),
+                new Capability( "bundler", "bundle" ),
+                new Capability( "clang" ),
+                new Capability( "cmake" ),
+                new Capability( "curl" ),
+                new Capability( "git" ),
+                new Capability( "gulp" ),
+                new Capability( "java" ),
+                new Capability( "JDK", "javac" ),
+                new Capability( "make" ),
+                new Capability( "maven", "mvn" ),
+                new Capability( "MSBuild", "xbuild" ),
+                new Capability( "node.js", "node" ),
+                new Capability( "node.js", "nodejs" ),
+                new Capability( "npm" ),
+                new Capability( "python" ),
```

Comment:
Yes yes, I got that. This PR is to get parity with xplat agent. Wanted to know what you have in mind, how are we  going to do the extendable capability model. Will there be a user provided json which tells us name of the capability and script path to run? or simply a key-value pair of json file?